### ISSUE TITLE
[3.10.0] GestureDetector/Inkwell onSecondaryTap

### ISSUE THREAD

**Issue by Sesa1988** [flutter/flutter#126780]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

If you use both, GestureDetector and Inkwell, onsecondaryTap doesn't trigger. 
To fix this you could move this down to Inkwell and it will work but using both should probably work as well?
This happens after upgrading to 3.10.0 on macOS.

1. Create a GestureDetector
2. Create Inwell as child
3. Trigger on secondary on macOS (tested on this platform) to open drawer

This also happens with to Inkwells and probably also other combinations.
**EDIT:** This also happens with ListTile and GestureDetector or Inkwell.

### Expected results

Drawer will open

### Actual results

Drawer doesn't open

### Code sample

<details><summary>Not working</summary>

```
GestureDetector(
          onSecondaryTap: () => _openMenu(context, _portfolio),
          child: InkWell(
            enableFeedback: true,
            onTap: () => _openMenu(context, _portfolio),
            onLongPress: () => _openMenu(context, _portfolio),
```
EDIT: just using this does also not work
```
InkWell(
      **onSecondaryTap: isFollowedTo == true ? () => _openMenu(context) : null,**
      child: ListTile(
        enableFeedback: true,
        onTap: isFollowedTo == true ? () => _openMenu(context) : null,
        onLongPress: isFollowedTo == true ? () => _openMenu(context) : null,
```

</details>

<details><summary>Partial fix, removed gesture detector and moved method</summary>

```
InkWell(
          enableFeedback: true,
          onTap: () => _openMenu(context, _portfolio),
          onLongPress: () => _openMenu(context, _portfolio),
          onSecondaryTap: () => _openMenu(context, _portfolio),
```
This is not fixable because Listtile does not have onsecondaryTap
```
InkWell(
      onSecondaryTap: isFollowedTo == true ? () => _openMenu(context) : null,
      child: ListTile(
        enableFeedback: true,
        onTap: isFollowedTo == true ? () => _openMenu(context) : null,
        onLongPress: isFollowedTo == true ? () => _openMenu(context) : null,
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Flutter Doctor output

[✓] Flutter (Channel stable, 3.10.0, on macOS 13.0 22A380 darwin-x64, locale en-DE)
    • Flutter version 3.10.0 on channel stable at /Users/user/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (5 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.92

[✓] Network resources
    • All expected network resources are available.

**Comment by huycozy** [comment]
I can see this issue appears on stable channel but it seems to be fixed on the latest master channel as I cannot reproduce it there.

<details>
<summary>Complete minimal sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(home: MyApp()));
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: GestureDetector(
          onSecondaryTap: () => _printST('onSecondaryTap'),
          child: InkWell(
            enableFeedback: true,
            onTap: () => _printST('onTap'),
            onLongPress: () => _printST('onLongPress'),
          ),
        ),
      ),
    );
  }

  _printST(String from) {
    print(from);
  }
}

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.10.0, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.10.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (35 hours ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.78.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.92

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.11.0-6.0.pre.63, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.11.0-6.0.pre.63 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7b8f7a2a64 (5 hours ago), 2023-05-14 17:42:29 -0400
    • Engine revision 326de1d849
    • Dart version 3.1.0 (build 3.1.0-110.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.78.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 113.0.5672.92

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

Please try switching to master channel and check this:
```
flutter channel master
flutter upgrade
```

I'll close this issue as fixed on the latest master channel. If you still experience this issue, kindly leave a comment below and I will reopen this. Thank you!

**Comment by huycozy** [comment]
It could be fixed by this PR: https://github.com/flutter/flutter/pull/119058

**Comment by Sesa1988** [comment]
Hi @huycozy , I can confirm, it's fixed on master. Is there any way to check for a timeline for a fix release to stable?

**Comment by huycozy** [comment]
@Sesa1988 You can read more about the build release channels here https://github.com/flutter/flutter/wiki/Flutter-build-release-channels.

**Comment by Sesa1988** [comment]
> It could be fixed by this PR: #119058

Im not sure if this would fix my issue with ListTile and InkWell onSEcondaryTap.

```
return InkWell(
      onSecondaryTap: isFollowedTo == true ? () => _openMenu(context) : null,
      child: ListTile(
        onTap: isFollowedTo == true ? () => _openMenu(context) : null,
        onLongPress: isFollowedTo == true ? () => _openMenu(context) : null,
```
SecondaryTap doesnt work here, mouse and trackpad.

**Comment by Sesa1988** [comment]
> It could be fixed by this PR: #119058

Does this contains also a fix for InkWell and ListTile wiht onSecondaryTap?

```
return InkWell(
      onSecondaryTap: isFollowedTo == true ? () => _openMenu(context) : null,
      child: ListTile(
        onTap: isFollowedTo == true ? () => _openMenu(context) : null,
        onLongPress: isFollowedTo == true ? () => _openMenu(context) : null,
```
